sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVInferSchema.scala

```diff
@@ -98,6 +98,11 @@ private[csv] object CSVInferSchema {
     }
   }
 
+  private def isInfOrNan(field: String, options: CSVOptions): Boolean = {
+    val infOrNan = Seq(options.nanValue, options.negativeInf, options.positiveInf)
+    infOrNan.contains(field)
```

Comment:
`field == options.nanValue || field == options.negativeInf || ...`,  to avoid creating a seq many times